specs/safe-liveness-checking.md

```diff
@@ -0,0 +1,66 @@
+# Safe Liveness Checking
+
+<!-- START doctoc generated TOC please keep comment here to allow auto update -->
+<!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE -->
+**Table of Contents**
+
+- [Overview](#overview)
+- [Liveness Checking Mechanism](#liveness-checking-mechanism)
+- [Liveness Check Implementation](#liveness-check-implementation)
+- [Liveness Check Validation](#liveness-check-validation)
+
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+
+## Liveness Checking Mechanism
+
+The Security Security Council uses a specially extended multisig...
+
+
+## Liveness checking methodology
+
+This is achieved using two types of contracts which the Safe contract has built-in support for:
+
+1. **Guard contracts:** can execute pre- and post- transaction checks.
+1. **Module contracts:** a contract which is added to the Safe by the signers, and thenceforth is
+   authorized to execute transactions via the Safe. This means the module must properly implement
+   auth conditions internally.
+
+### The Liveness Guard
+
+For implementing liveness checks a `LivenessGuard` is created which receives the signatures from
+each executed transaction, and tracks the latest time at which a transaction was signed by each
+signer. This time is made publicly available by calling a `lastSigned(address)(Timestamp)` method.
+
+Signers may also call the contract directly in order to prove liveness.
```

Comment:
Nit, might be nice to include the method name here with the same syntax as `lastSigned(address)(Timestamp)` above (or ignore the parentheses if there's no args, not sure what the usual syntax used in this repo is)
```suggestion
Signers may also call the contract's `showLiveness()()` method directly in order to prove liveness.
```

Reply:
There is a tradeoff with baking in the ABI, it can go stale. Previously a lot of the specs went stale in certain ways but this doesn't seem like it is something that will go stale

Reply:
It looks like other places in this file it uses functions so using function names is fine

Reply:
Yes, I don't think this spec will evolve too quickly/often once finalized, so it should be OK to include function names.